src/equations/3d/hyperbolic_diffusion.jl

```diff
@@ -217,6 +232,18 @@ function source_terms_harmonic(ut, u, x, element_id, t, n_nodes, equation::Hyper
   return nothing
 end
 
+@inline function source_terms_harmonic(u, x, t, equations::HyperbolicDiffusionEquations3D)
+  # harmonic solution ϕ = (sinh(πx)sin(πy) + sinh(πy)sin(πx))/sinh(π), so f = 0
+  inv_Tr = inv(equations.Tr)
```

Comment:
Should we just store `inv_Tr` with the system of equations instead? It's a constant and it seems like its inversion is required in all source terms. If it's too much for this PR, let's create an issue.

Reply:
Good suggestion. Done in bcb3be7.

Reply:
:+1: